source/pdal/pdalc_pipeline.cpp

```diff
@@ -222,80 +217,122 @@ extern "C"
     {
         Pipeline *ptr = reinterpret_cast<Pipeline *>(pipeline);
 
-        if (ptr && ptr->get())
+        try
         {
-            try
-            {
-                ptr->get()->setLogLevel(level);
-            }
-            catch (const std::exception &e)
-            {
-                printf("Found error while setting log level: %s\n", e.what());
-            }
+            if (level < 0 || level > 8)
+                throw pdal_error("log level must be between 0 and 8!");
+
+            pdal::LogPtr lptr = ptr->manager->log();
+            lptr->setLevel(static_cast<pdal::LogLevel>(level));
+        }
+        catch (const std::exception &e)
+        {
+            printf("Found error while setting log level: %s\n", e.what());
         }
+
     }
 
     int PDALGetPipelineLogLevel(PDALPipelinePtr pipeline)
     {
         Pipeline *ptr = reinterpret_cast<Pipeline *>(pipeline);
-        return (ptr && ptr->get()) ? ptr->get()->getLogLevel() : 0;
+        try
+        {
+            return (ptr)
+                   ? static_cast<int>(
+                       ptr->manager->log()->getLevel()
+                   ) : 0;
+        }
+        catch (const std::exception &e)
+        {
+            printf("Found error while getting log level: %s\n", e.what());
```

Comment:
e.what().c_str() ? Or just direct to cout?

Reply:
as above - I am not sure this is neccesary.